src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -463,5 +463,29 @@ internal static bool IsVoidOrUnit(Type type)
 
             return type == typeof(void) || type.FullName == "Microsoft.FSharp.Core.Unit";
         }
+
+        /// <summary>
+        /// Returns the invokable method for a given named indexer.
+        /// </summary>
+        /// <param name="type">Type to reflect on for indexer.</param>
+        /// <param name="indexerTypes">List of indexer types that matches the indexer type order.</param>
+        /// <returns>Invokable indexer method</returns>
+        public static MethodInfo? GetDefaultIndexer(Type type, Type[] indexerTypes)
+        {
+            const BindingFlags bindingFlags = BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance;
```

Comment:
Completely off topic, but one of the things being considered for future C# is the ability to target-type the name lookup so that you can do `type.GetProperty(indexerName, NonPublic | Public | Instance)` etc.